### PR TITLE
Delete widgets and support for tooltips

### PR DESCRIPTION
**Pull request by Jacob-Grafenstein**
This PR does several things:
1. Delete widgets from the dashboard using the x icon
2. Create a tooltip component
3. Use the tooltip component in the sidebar (a known bug exists where if you swipe over the list at fast speeds, the tooltips stay visible). In fact, this bug is PERVASIVE: https://github.com/facebook/react/issues/6807 https://github.com/facebook/react/issues/4492
4. Styles the scrollbars in the widgets

### CODE REVIEW

**Comment by Jacob-Grafenstein** [comment]
@andriyrusyn @RobcPeng please review this PR when you get the chance

File: client/src/App.scss [hunk 1]
```diff
@@ -125,4 +125,24 @@ button {
...
+
+/* Handle */
+::-webkit-scrollbar-thumb {
+  background: #888;
+}
+
+/* Handle on hover */
+::-webkit-scrollbar-thumb:hover {
+  background: #555;
+}
```
Comment (RobcPeng) [inline client/src/App.scss]: Does this need a newline after it?

File: client/src/components/mainContentArea/mainContent.jsx [hunk 2]
```diff
@@ -32,7 +46,7 @@ export default class MainContent extends React.Component {
     return _.map(this.props.lg, function(l, i) {
       return (
         <div key={i} className={classNames(l.static ? "static" : "")} data-grid={l}>
-          <div className="wrapper">{widgetFriendlyNameMap[widgetList[i]]}</div>
+          <div className="wrapper">{getWidgetForFriendlyName(widgetList[i], i)}</div>
         </div>
       );
     });
```
Comment (RobcPeng) [inline client/src/components/mainContentArea/mainContent.jsx]: should the }); be on different lines?

File: client/src/components/popperWrapper/popperWrapper.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                {poppedComponent}
+              </div>
+            )}
+          </Popper>,
+          document.body
+        )}
+      </Manager>
+    )
+  }
+}
```
Comment (RobcPeng) [inline client/src/components/popperWrapper/popperWrapper.jsx]: Newline probably

File: client/src/App.scss (hunk 1 above)
Comment (Jacob-Grafenstein) [reply client/src/App.scss]: It's got one, otherwise a little red circle will appear on the last line

File: client/src/actions/layoutActions.js [hunk 4]
```diff
@@ -79,4 +79,10 @@ export function addWidgetToLayout(widgetType) {
   return dispatch => {
     dispatch({ type: ADD_WIDGET_TO_LAYOUT, widgetType });
   }
-}
```
Comment (Jacob-Grafenstein) [inline client/src/actions/layoutActions.js]: @RobcPeng for example, right here

File: client/src/components/popperWrapper/popperWrapper.jsx (hunk 3 above)
Comment (Jacob-Grafenstein) [reply client/src/components/popperWrapper/popperWrapper.jsx]: ones already there :)

File: client/src/components/mainContentArea/mainContent.jsx (hunk 2 above)
Comment (Jacob-Grafenstein) [reply client/src/components/mainContentArea/mainContent.jsx]: So, the `{}` basically just means `I want to run some javascript in this HTML code`, so it's totally ok and generally accepted to keep the brackets right up against the function you want to call. That said, I can totally see how moving `{getWidgetForFriendlyName(widgetList[i], i)}` to its own line will help with readability, so I will go ahead and do that.

**Pull request merged**